src/core/dappletContext.ts

```diff
@@ -20,20 +21,70 @@ export class DappletContext {
 
     constructor(config: ContextConfig = DEFAULT_CONFIG) {
         config = { ...DEFAULT_CONFIG, ...config };
-        this.dappletProviders = config.providers!;
+        this.dappletProviders = config.providers! || [];
         this.featureRegistry = new FeatureRegistry(config.features || []);
     }
 
     async processRequest(request: DappletRequest): Promise<DappletTxResult> {
+        // dapplet loading and validation
+        for (let i = 0; i < request.frames.length; i++) {
+            const dapplet = await this._loadDapplet(request.frames[i].dappletId);
+            this._validateDapplet(dapplet);
+            request.frames[i].dapplet = dapplet;
+        }
```

Comment:
Does it mean, all dapplets get loaded in sequence and not in parallel?
not good for performance.

suggest something like:
```
dapplets[]=await Promise.all(
   request.frames.map(f=>
       this._loadDapplet(f.dappletId)
            .then(d=>this._validate(d))
            .catch(????)
       )
);
```